### ISSUE TITLE
Error deploying Function App to Azure when custom Task label contains certain characters

### ISSUE THREAD

**Issue by e14mattc** [microsoft/vscode-azurefunctions#1496]
Repro steps:

1. Open tasks.json
2. Define a task with a label that contains '++', e.g. 'Open in Notepad++'. Full task definition example;

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Open in Notepad++",
            "type": "shell",
            "command": "c:\\progra~2\\notepa~1\\notepad++.exe \"${file}\"",
            "problemMatcher": []
        }
    ]
}
```
3. Select a Function, and select 'Deploy to Function App...'
4. It'll quickly error out with this message...

![image](https://user-images.githubusercontent.com/9819307/65144033-50b5cd80-da0e-11e9-852a-0fc6fdaa0caa.png)

Action: azureFunctions.deploy
Error type: SyntaxError
Error Message: Invalid regular expression: /^(Workspace: )?Open in Notepad++$/: Nothing to repeat

Version: 0.18.1
OS: win32
Product: Visual Studio Code
Product Version: 1.38.1
Language: en

<details>
<summary>Call Stack</summary>

```
extension.bundle.js:331:355656extension.bundle.js:331:355656
module.exports.r.find.e extension.bundle.js:331:355831
Object.<anonymous> extension.bundle.js:331:355628
o extension.bundle.js:331:354922
```

</details>

**Comment by EricJizbaMSFT** [comment]
@e14mattc thanks for filing - we'll look into it. In the meantime I assume you're not blocked? It works if you just remove the special characters?

**Comment by e14mattc** [comment]
Yeah, I'm all good .. once I figured out it was just the ++ characters causing the problem, I just removed them from the label. Thanks for following up.

**Comment by EricJizbaMSFT** [comment]
@e14mattc we just released v0.19.0 of the extension with this fix. Feel free to add back all the pluses to your task label that you want!
